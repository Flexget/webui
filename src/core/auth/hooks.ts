export const useAuth = () => {
  const [isLoggedIn, setIsLoggedIn] = useState(false);

  let login = () => 
}
