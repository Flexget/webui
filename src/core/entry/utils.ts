import { RawEntry, FieldNames, MappingType, GettersType, BaseEntry } from './types';
import { MovieEntry, movieFieldList, RawMovieEntry, MovieFieldNames } from './fields/movies';
import { SeriesEntry, seriesFieldList, SeriesFieldNames, RawSeriesEntry } from './fields/series';
import {
  EpisodeEntry,
  RawEpisodeEntry,
  episodesFieldList,
  EpisodeFieldNames,
} from './fields/episodes';

export const isMovie = (entry: RawEntry): entry is RawMovieEntry => !!entry.movieName;
export const isSeries = (entry: RawEntry): entry is RawSeriesEntry =>
  !!entry.seriesName && !entry.seriesSeason;
export const isEpisode = (entry: RawEntry): entry is RawEpisodeEntry & SeriesEntry =>
  !!(entry.seriesSeason && entry.seriesEpisode && entry.seriesName);

const getPropFn = <U>(mapping: MappingType<U>) => {
  const mapByField = mapping.reduce<Partial<Record<FieldNames<U>, string[]>>>(
    (res, item) =>
      Object.entries(item).reduce(
        (out, [k, v]) => ({
          ...out,
          [k]: [...(out[k] ?? []), v],
        }),
        res,
      ),
    {},
  );
  return (entry: U, field: FieldNames<U>) =>
    (mapByField[field] ?? ([] as string[])).reduce((res, val) => res ?? entry[val], undefined);
};

const makeGetters = <T>(names: Record<string, FieldNames<T>>, mapping: MappingType<T>) => {
  const getProp = getPropFn<T>(mapping);
  return (entry: T) =>
    Object.values(names).reduce(
      (obj, key: FieldNames<T>) => ({
        ...obj,
        get [key]() {
          return getProp(entry, key);
        },
      }),
      {} as GettersType<T>,
    );
};

const getMovieProps = makeGetters<RawMovieEntry>(MovieFieldNames, movieFieldList);
const getSeriesProps = makeGetters<RawSeriesEntry>(SeriesFieldNames, seriesFieldList);
const getEpisodeProps = makeGetters<RawEpisodeEntry>(EpisodeFieldNames, episodesFieldList);

export const toMovieEntry = (entry: RawMovieEntry): MovieEntry => {
  return {
    ...entry,
    ...getMovieProps(entry),
  };
};

export const toSeriesEntry = (entry: RawSeriesEntry): SeriesEntry => ({
  ...entry,
  ...getSeriesProps(entry),
});

export const toEpisodeEntry = (entry: RawEpisodeEntry): EpisodeEntry => ({
  ...entry,
  ...getEpisodeProps(entry),
});

export const toDefaultEntry = (entry: BaseEntry): BaseEntry => ({
  ...entry,
});
