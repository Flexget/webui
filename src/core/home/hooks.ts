import { useMemo } from 'react';
import { useContainer } from 'unstated-next';
import { PluginContainer } from 'core/plugins/hooks';
import { CardInfo } from './types';

export const useGetHomeCards = () => {
  const { pluginMap } = useContainer(PluginContainer);
  const cards: CardInfo[] = useMemo(
    () =>
      Object.entries(pluginMap).flatMap(([path, { cardComponent, routeDisplayName }]) =>
        cardComponent
          ? [
              {
                path,
                component: cardComponent,
                name: routeDisplayName,
              },
            ]
          : [],
      ),
    [pluginMap],
  );
  return { cards };
};
