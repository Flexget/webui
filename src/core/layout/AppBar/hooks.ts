import { useState, useEffect, useCallback, ComponentType } from 'react';
import { createContainer, useContainer } from 'unstated-next';
import { useFlexgetAPI } from 'core/api';
import { Method } from 'utils/fetch';
import { Operation } from './types';

interface IconProps {
  name: string;
  onClick: () => void;
  Icon: ComponentType;
}

export interface ContextualProps {
  icons?: IconProps[];
  title?: string;
  onClose?: () => void;
}

const defaultTitle = 'Flexget Manager';

export const AppBarContainer = createContainer(() => {
  const [title, setTitle] = useState(defaultTitle);
  const [content, setContent] = useState<JSX.Element>();
  const [contextualProps, setContextualProps] = useState<ContextualProps>();
  const [contextualMode, setContextual] = useState(false);

  return [
    { title, content, contextualProps, contextualMode },
    { setTitle, setContent, setContextualProps, setContextual },
  ] as const;
});

export const useInjectPageTitle = (title: string) => {
  const [, { setTitle }] = useContainer(AppBarContainer);

  const clear = useCallback(() => {
    setTitle(defaultTitle);
    document.title = defaultTitle;
  }, [setTitle]);

  useEffect(() => {
    setTitle(title);
    document.title = `${defaultTitle} - ${title}`;
  }, [setTitle, title]);

  return clear;
};

export const useInjectContent = (content: JSX.Element) => {
  const [, { setContent }] = useContainer(AppBarContainer);

  const clear = useCallback(() => setContent(undefined), [setContent]);

  useEffect(() => {
    setContent(content);

    return clear;
  }, [clear, content, setContent]);

  return clear;
};

export const useContextualAppBar = (props?: ContextualProps) => {
  const [, { setContextual, setContextualProps }] = useContainer(AppBarContainer);

  const clear = useCallback(() => {
    setContextualProps(undefined);
    setContextual(false);
  }, [setContextual, setContextualProps]);

  useEffect(() => {
    setContextualProps(props);

    return clear;
  }, [clear, props, setContextualProps]);

  return { setContextual, clear };
};

export const useServerOperation = (operation: Operation, onSuccess = () => {}) => {
  const [state, request] = useFlexgetAPI('/server/manage', Method.Post);
  const makeRequest = useCallback(async () => {
    const resp = await request({ operation });
    if (resp.ok) {
      onSuccess();
    }
  }, [onSuccess, operation, request]);

  return [state, makeRequest] as const;
};
