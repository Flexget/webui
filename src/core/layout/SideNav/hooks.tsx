import { useEffect, useState, useCallback } from 'react';
import { useFlexgetAPI } from 'core/api';
import { createContainer } from 'unstated-next';
import { AuthContainer } from 'core/auth/container';
import { useGlobalStatus } from 'core/status/hooks';
import { Method } from 'utils/fetch';

interface VersionResponse {
  apiVersion: string;
  flexgetVersion: string;
  latestVersion: string;
}

export const VersionContainer = createContainer(() => {
  const [loggedIn] = AuthContainer.useContainer();
  const [version, setVersion] = useState<VersionResponse>();
  const [{ loading, error }, getVersion] = useFlexgetAPI<VersionResponse>('/server/version');
  useGlobalStatus(loading, error);

  useEffect(() => {
    const fn = async () => {
      if (!loggedIn) {
        const resp = await getVersion();
        if (resp.ok) {
          setVersion(resp.data);
        }
      }
    };
    fn();
  }, [getVersion, loggedIn]);

  return { loading, version };
});

export const enum ServerOperation {
  Reload = 'reload',
  Shutdown = 'shutdown',
}

export const useServerOperation = (operation: ServerOperation, onSuccess = () => {}) => {
  const [state, request] = useFlexgetAPI('/server/manage', Method.Post);
  const makeRequest = useCallback(async () => {
    const resp = await request({ operation });
    if (resp.ok) {
      onSuccess();
    }
  }, [onSuccess, operation, request]);

  return [state, makeRequest] as const;
};

export const useGetPlugins = () => {
  const [plugins, setPlugins] = useState<string[]>([]);
  const [{ loading }, getPlugins] = useFlexgetAPI<string[]>('/database/plugins');

  useEffect(() => {
    const fn = async () => {
      const resp = await getPlugins();
      if (resp.ok) {
        setPlugins(resp.data);
      }
    };
    fn();
  }, [getPlugins]);

  return { loading, plugins };
};
