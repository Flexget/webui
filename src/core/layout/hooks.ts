import { createContainer } from 'unstated-next';
import { useOverlayState } from 'utils/hooks';

export const OperationsContainer = createContainer(useOverlayState);
