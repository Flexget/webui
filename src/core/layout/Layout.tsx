import React, { useCallback } from 'react';
import ErrorStatus from 'core/status/ErrorStatus';
import InfoStatus from 'core/status/InfoStatus';
import LoadingBar from 'core/status/LoadingBar';
import { useMediaQuery, useTheme } from '@material-ui/core';
import { useOverlayState } from 'utils/hooks';
import Logo from './Logo';
import Navbar from './Navbar';
import SideNav from './SideNav';
import {
  header,
  wrapper,
  logoWrapper,
  nav,
  main,
  sidebar,
  contentWithSidebar,
  content,
} from './styles';

const Layout: React.FC = ({ children }) => {
  const theme = useTheme();
  const matches = useMediaQuery(theme.breakpoints.down('sm'));
  const [sidebarOpen, { toggle, close }] = useOverlayState(matches);

  const contentCssFn = useCallback(() => contentWithSidebar(sidebarOpen), [sidebarOpen]);

  return (
    <div css={wrapper}>
      <header css={header}>
        <div css={logoWrapper}>
          <Logo sidebarOpen={sidebarOpen} />
        </div>
        <nav css={nav}>
          <Navbar toggleSidebar={toggle} />
          <LoadingBar />
        </nav>
      </header>
      <main css={main}>
        <aside css={sidebar}>
          <SideNav sidebarOpen={sidebarOpen} onClose={close} />
        </aside>
        <section css={[content, contentCssFn()]}>{children}</section>
        <ErrorStatus />
        <InfoStatus />
      </main>
    </div>
  );
};

export default Layout;
