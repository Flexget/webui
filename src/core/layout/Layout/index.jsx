import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Logo from 'core/layout/Logo';
import Navbar from 'core/layout/Navbar';
import SideNav from 'core/layout/SideNav';
import LoadingBar from 'common/LoadingBar';
import ErrorStatus from 'common/ErrorStatus';
import InfoStatus from 'common/InfoStatus';
import { Wrapper, Main, Header, LogoWrapper, Nav, SideBar, Content } from './styles';

class Layout extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
  };

  state = {
    sideBarOpen: (window.matchMedia && !!window.matchMedia('(min-width: 600px)').matches) || false,
  };

  toggleSideBar = () => {
    this.setState(({ sideBarOpen }) => ({
      sideBarOpen: !sideBarOpen,
    }));
  };

  render() {
    const { children } = this.props;
    const { sideBarOpen } = this.state;

    return (
      <Wrapper>
        <Header>
          <LogoWrapper>
            <Logo sideBarOpen={sideBarOpen} />
          </LogoWrapper>
          <Nav>
            <Navbar toggle={this.toggleSideBar} />
            <LoadingBar />
          </Nav>
        </Header>
        <Main>
          <SideBar>
            <SideNav sideBarOpen={sideBarOpen} toggle={this.toggleSideBar} />
          </SideBar>
          <Content open={sideBarOpen}>{children}</Content>
          <ErrorStatus />
          <InfoStatus />
        </Main>
      </Wrapper>
    );
  }
}

export default Layout;
