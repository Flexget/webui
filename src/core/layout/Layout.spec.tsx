import React from 'react';
import { FetchMock } from 'jest-fetch-mock';
import { act, create, ReactTestRenderer } from 'react-test-renderer';
import { themed, router, provider, authProvider, statusProvider } from 'utils/tests';
import { RouteContainer } from 'core/routes/hooks';
import Layout from './Layout';

const fetchMock = fetch as FetchMock;

function renderLayout() {
  return provider(
    statusProvider(
      authProvider(
        router(
          themed(
            <RouteContainer.Provider>
              <Layout>
                <div />
              </Layout>
            </RouteContainer.Provider>,
          ),
        ),
      ),
    ),
    {
      router: { location: {} },
      version: {},
      status: { loading: {} },
      routes: {},
    },
  );
}
describe('common/layout', () => {
  beforeEach(() => {
    fetchMock.mockResponseOnce(
      JSON.stringify({
        apiVersion: '1.1.2',
        flexgetVersion: '2.10.60',
        latestVersion: '2.10.60',
      }),
    );
  });

  it('renders correctly', async () => {
    let tree: ReactTestRenderer | undefined;
    await act(async () => {
      tree = create(renderLayout());
    });
    expect(tree?.toJSON()).toMatchSnapshot();
  });
});
