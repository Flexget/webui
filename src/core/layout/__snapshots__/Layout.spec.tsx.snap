// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/layout renders correctly 1`] = `
.emotion-6 {
  position: static;
}

.emotion-5 {
  background-color: #ef6c00;
  color: #fff;
  min-height: 5rem;
}

.emotion-0 {
  color: #fff;
}

.emotion-1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-11 .MuiDrawer-paper {
  position: relative;
  background-color: #263238;
  height: calc(100vh - 10rem);
  width: 0;
  -webkit-transition: width 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: width 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  border-right: none !important;
}

@media (min-width:600px) {
  .emotion-11 .MuiDrawer-paper {
    height: calc(100vh - 5rem);
    width: 6rem;
  }
}

.emotion-10 {
  color: #ef6c00;
}

.emotion-7 {
  margin: 0;
}

<div
  css={
    Object {
      "map": undefined,
      "name": "oh8avo",
      "next": undefined,
      "styles": "
  display: flex;
  flex-direction: column;
  height: 100%;
  min-height: 100vh;
",
      "toString": [Function],
    }
  }
>
  <header
    css={
      Object {
        "map": undefined,
        "name": "1mfkpn7",
        "next": undefined,
        "styles": "
  display: flex;
  min-height: 5rem;
  flex-direction: column;
  z-index: 2;
  position: fixed;
  width: 100%;
  @media (min-width:600px) {
    flex-direction: row;
  }
",
        "toString": [Function],
      }
    }
  >
    <div
      css={
        Object {
          "map": undefined,
          "name": "1r0czf3",
          "next": undefined,
          "styles": "
  height: 5rem;
",
          "toString": [Function],
        }
      }
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <div
          css={
            Array [
              Object {
                "map": undefined,
                "name": "1w2nirn",
                "next": undefined,
                "styles": "
  background: #263238 url(test-file-stub) no-repeat center;
  background-size: 17.5rem;
  height: 100%;
  transition: width 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,background-size 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
",
                "toString": [Function],
              },
              Object {
                "map": undefined,
                "name": "1af4v5",
                "next": undefined,
                "styles": "
  @media (min-width:600px) {
    width: 6rem;
    background-size: 25rem;
  }
",
                "toString": [Function],
              },
            ]
          }
        />
      </a>
    </div>
    <nav
      css={
        Object {
          "map": undefined,
          "name": "jdqp6j",
          "next": undefined,
          "styles": "
  height: 5rem;
  flex: 1;
",
          "toString": [Function],
        }
      }
    >
      <header
        className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed emotion-6 MuiAppBar-colorPrimary mui-fixed"
      >
        <div
          className="MuiToolbar-root MuiToolbar-regular emotion-5 MuiToolbar-gutters"
        >
          <button
            className="MuiButtonBase-root MuiIconButton-root emotion-0"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
          <h6
            className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit"
          />
          <div
            className="emotion-1"
          />
          <a
            href="/config"
            onClick={[Function]}
          >
            <button
              aria-label="Config"
              className="MuiButtonBase-root MuiIconButton-root emotion-0"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM5.92 19H5v-.92l9.06-9.06.92.92L5.92 19zM20.71 5.63l-2.34-2.34c-.2-.2-.45-.29-.71-.29s-.51.1-.7.29l-1.83 1.83 3.75 3.75 1.83-1.83c.39-.39.39-1.02 0-1.41z"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </button>
          </a>
          <a
            href="/log"
            onClick={[Function]}
          >
            <button
              aria-label="Log"
              className="MuiButtonBase-root MuiIconButton-root emotion-0"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m1.1-2H3.9c-.5 0-.9.4-.9.9v16.2c0 .4.4.9.9.9h16.2c.4 0 .9-.5.9-.9V3.9c0-.5-.5-.9-.9-.9zM11 7h6v2h-6V7zm0 4h6v2h-6v-2zm0 4h6v2h-6zM7 7h2v2H7zm0 4h2v2H7zm0 4h2v2H7z"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </button>
          </a>
          <button
            aria-label="Manage"
            className="MuiButtonBase-root MuiIconButton-root emotion-0"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M0 0h20v20H0V0z"
                  fill="none"
                  transform="scale(1.2, 1.2)"
                />
                <path
                  d="M15.95 10.78c.03-.25.05-.51.05-.78s-.02-.53-.06-.78l1.69-1.32c.15-.12.19-.34.1-.51l-1.6-2.77c-.1-.18-.31-.24-.49-.18l-1.99.8c-.42-.32-.86-.58-1.35-.78L12 2.34c-.03-.2-.2-.34-.4-.34H8.4c-.2 0-.36.14-.39.34l-.3 2.12c-.49.2-.94.47-1.35.78l-1.99-.8c-.18-.07-.39 0-.49.18l-1.6 2.77c-.1.18-.06.39.1.51l1.69 1.32c-.04.25-.07.52-.07.78s.02.53.06.78L2.37 12.1c-.15.12-.19.34-.1.51l1.6 2.77c.1.18.31.24.49.18l1.99-.8c.42.32.86.58 1.35.78l.3 2.12c.04.2.2.34.4.34h3.2c.2 0 .37-.14.39-.34l.3-2.12c.49-.2.94-.47 1.35-.78l1.99.8c.18.07.39 0 .49-.18l1.6-2.77c.1-.18.06-.39-.1-.51l-1.67-1.32zM10 13c-1.65 0-3-1.35-3-3s1.35-3 3-3 3 1.35 3 3-1.35 3-3 3z"
                  transform="scale(1.2, 1.2)"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
      </header>
    </nav>
  </header>
  <main
    css={
      Object {
        "map": undefined,
        "name": "fpd4ie",
        "next": undefined,
        "styles": "
  display: flex;
  flex-direction: row;
  padding-top: 10rem;
  flex: 1;
  max-height: 100vh;
  @media (min-width:600px) {
    padding-top: 5rem;
  }
",
        "toString": [Function],
      }
    }
  >
    <aside
      css={
        Object {
          "map": undefined,
          "name": "qdovln",
          "next": undefined,
          "styles": "
  overflow-y: auto;
",
          "toString": [Function],
        }
      }
    >
      <div
        className="MuiDrawer-root MuiDrawer-docked emotion-11"
      >
        <div
          className="MuiPaper-root MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft"
        >
          <div
            css={
              Object {
                "map": undefined,
                "name": "mrqyz6",
                "next": undefined,
                "styles": "
  width: inherit;
  height: inherit;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
",
                "toString": [Function],
              }
            }
          >
            <ul
              className="MuiList-root MuiList-padding"
              css={
                Object {
                  "map": undefined,
                  "name": "1nxb9gk",
                  "next": undefined,
                  "styles": "
            width: inherit;
          ",
                  "toString": [Function],
                }
              }
            />
            <div
              className="emotion-10"
            >
              <p
                className="MuiTypography-root emotion-7 MuiTypography-body1"
              >
                Version Info
              </p>
              <p
                className="MuiTypography-root emotion-7 MuiTypography-body1"
              >
                Flexget: 2.10.60 
                <a
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall"
                  href="https://flexget.com/ChangeLog"
                  onBlur={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M11 18h2v-2h-2v2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </a>
              </p>
              <p
                className="MuiTypography-root emotion-7 MuiTypography-body1"
              >
                API: 1.1.2
              </p>
            </div>
          </div>
        </div>
      </div>
    </aside>
    <section
      css={
        Array [
          Object {
            "map": undefined,
            "name": "12lzmjs",
            "next": undefined,
            "styles": "
  flex: 1;
  overflow-y: auto;
  padding: 1.6rem;
  opacity: 1;
  transition: opacity 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,margin-left 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;

  @media (max-width:959.95px) {
    margin-left: 0;
    padding: 0.8rem;
  }
",
            "toString": [Function],
          },
          Object {
            "map": undefined,
            "name": "leo25p",
            "next": undefined,
            "styles": "
  @media (max-width:959.95px) {
    opacity: 1;
    display: block;
  }
",
            "toString": [Function],
          },
        ]
      }
    >
      <div />
    </section>
  </main>
</div>
`;
